src/http/controllers/core.controller.ts

```diff
@@ -157,4 +167,25 @@ export class CoreController {
             this._messageGateway._server.to(data.channel).emit(data.event, data.data);
         }
     }
+
+    @MessagePattern('updateMetric')
+    async updateMetric(@Payload() data: { name: string; value: number }): Promise<void> {
+        if (data.name == MetricNames.LUM_CURRENT_SUPPLY) {
+            await this._lumCurrentSupply.set(data.value);
+        } else if (data.name == MetricNames.DFRACT_CURRENT_SUPPLY) {
+            await this._dfrCurrentSupply.set(data.value);
+        } else if (data.name == MetricNames.COMMUNITY_POOL_SUPPLY) {
+            await this._communityPoolSupply.set(data.value);
+        } else if (data.name == MetricNames.LUM_PRICE_USD) {
+            await this._lumPriceUSD.set(data.value);
+        } else if (data.name == MetricNames.LUM_PRICE_EUR) {
+            await this._lumPriceEUR.set(data.value);
+        } else if (data.name == MetricNames.DFRACT_MA_BALANCE) {
+            await this._dfrMaBalance.set(data.value);
+        } else if (data.name == MetricNames.MARKET_CAP) {
+            await this._marketCap.set(data.value);
+        } else if (data.name == MetricNames.TWITTER_FOLLOWERS) {
+            await this._twitterFollowers.set(data.value);
+        }
+    }
```

Comment:
Would it make it make a bit more readable to do it in the object literal way ?

```
async updateMetric = @Payload() data: { name: string; value: number }): Promise<void> =>
    ({
        [MetricNames.LUM_CURRENT_SUPPLY]: await this._lumCurrentSupply.set(data.value),
        [MetricNames.DFRACT_CURRENT_SUPPLY]: await this._dfrCurrentSupply.set(data.value),
        [MetricNames.COMMUNITY_POOL_SUPPLY]: await this._communityPoolSupply.set(data.value),
        ...
    }[data[name] || defaultValue);
```

Reply:
Not possible using NestJS with the MessagePattern syntax